llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1729,6 +1729,36 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
     }
     break;
   }
+  case TargetOpcode::G_STEP_VECTOR: {
+    if (!MI->getOperand(1).isCImm()) {
+      report("operand must be cimm", MI);
+      break;
+    }
+
+    if (!MI->getOperand(1).getCImm()->getValue().isStrictlyPositive()) {
+      report("step must be >= 0", MI);
+      break;
+    }
```

Comment:
If the step is unsigned as the doc specifies, then does this still make sense?

Reply:
The step is `ConstantInt` or `APInt`. Neither enforces the behaviour.

Reply:
For step=0, we create a step vector of all zeros. The canonical representation is a splat vector of zeros. It would loose the stepping property.

Reply:
So the message should say `> 0` not `>= 0`?